models/user/badge.go

```diff
@@ -42,13 +46,27 @@ func GetUserBadges(ctx context.Context, u *User) ([]*Badge, int64, error) {
 	return badges, count, err
 }
 
+// GetBadgeUsers returns the users that have a specific badge.
+func GetBadgeUsers(ctx context.Context, b *Badge) ([]*User, int64, error) {
```

Comment:
I think pagination is necessary here because you don't know how many users will have the badges which maybe a big number.